### PR TITLE
Add scaleX and scaleY Parameters in Transform.Scale.

### PR DESCRIPTION
**Pull request by nayeemtby**
Added scaleX and scaleY parameters in Transform.Scale. The scale parameter acts the same by overriding scaleX and scaleY.
Also edited the Tooltip help.

Addresses feature request #92342 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/basic.dart
Comment (nayeemtby) [reply packages/flutter/lib/src/widgets/basic.dart]: Fixed

**Comment by nayeemtby** [comment]
Can all of the commits be squashed into one? I got caught in the checkings.

**Comment by Hixie** [comment]
> Can all of the commits be squashed into one? I got caught in the checkings.

Yes, this will happen automatically when we land it. You can also do it manually using `git rebase -i`.

**Comment by nayeemtby** [comment]
> > Can all of the commits be squashed into one? I got caught in the checkings.
> 
> Yes, this will happen automatically when we land it. You can also do it manually using `git rebase -i`.



> > Can all of the commits be squashed into one? I got caught in the checkings.
> 
> Yes, this will happen automatically when we land it. You can also do it manually using `git rebase -i`.

That's nice. But I cannot get much info on this luci-flutter test failure. Where can I get some documentation or know what went wrong?

File: packages/flutter/lib/src/widgets/basic.dart
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart:1271]: You can remove "Please note that" per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose

Avoid "you" per https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-the-passive-voice-recommend-do-not-require

Comment (Hixie) [reply packages/flutter/lib/src/widgets/basic.dart:1271]: I would say something like:

> Creates a widget that scales its child along the 2D plane.
>
> The `scaleX` argument provides the scalar by which to multiply the `x` axis, and the `scaleY` argument provides the scalar by which to multiply the `y` axis. Either may be omitted, in which case that axis defaults to 1.0.
>
> For convenience, instead of providing `scaleX` and `scaleY`, the `scale` parameter may be used.
>
> One of `scale`, `scaleX`, and `scaleY` must be non-null. If `scale` is provided, the other two must be null; similarly,  if it is not provided, one of the other two must be.

File: packages/flutter/lib/src/widgets/basic.dart
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart:1307]: No need to say "In Transform.scale()", the exception will include a stack trace which makes that unambiguous.

Avoid "you" for the same reason as cited above. I would just say "At least one of 'scale', 'scaleX', and 'scaleY' must be provided.".

File: packages/flutter/lib/src/widgets/basic.dart
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart:1308]: Here you can say "If 'scale' is specified, 'scaleX', and 'scaleY' must both be null.".

**Comment by Hixie** [comment]
Just some minor issues with the docs and assert text, but otherwise LGTM.

**Review by Hixie — APPROVED** [review]
LGTM modulo minor issues discussed above

**Comment by Hixie** [comment]
(once you've made the changes please ping #hackers on [discord](https://github.com/flutter/flutter/wiki/Chat) to ask if someone can give you the second review. thanks!)

**Comment by Hixie** [comment]
(You can ignore the 'luci-flutter' test failure, that's just telling us the tree is broken right now and is unrelated to your PR.)

File: packages/flutter/lib/src/widgets/basic.dart
Comment (nayeemtby) [reply packages/flutter/lib/src/widgets/basic.dart:1271]: Can I copy this whole thing? Cause it's like simply better.

Comment (nayeemtby) [reply packages/flutter/lib/src/widgets/basic.dart:1271]: Can I copy this whole thing? This looks better.

File: packages/flutter/lib/src/widgets/basic.dart
Comment (flar) [inline packages/flutter/lib/src/widgets/basic.dart:1271]: "must be" at the end of the sentence can either refer to "provided" or "null" depending on how the reader interprets the antecedent.

If A is X, then B must by Y,
    otherwise
if A is !X, then B must be ?

The reader could imply the missing term comes from the previous statement, or from the earlier phrase in the current statement. I'd add "must be provided" or "must be non-null".

**Review by flar — COMMENTED** [review]
LGTM other than the doc comment nit.

File: packages/flutter/lib/src/widgets/basic.dart
Comment (flar) [reply packages/flutter/lib/src/widgets/basic.dart]: I commented in another review, but leaving the tail end as "must be" leaves the reader wondering whether you are referring to "must be provided" or "must be null". There are 2 extremely similar statements there and the antecedent could come either from the other part of the same phrase or the analogous part of the previous phrase - creating ambiguity.

File: packages/flutter/lib/src/widgets/basic.dart
Comment (nayeemtby) [reply packages/flutter/lib/src/widgets/basic.dart]: Done. But one check failed maybe because of internal error. Can you re-trigger the check?

Comment (flar) [reply packages/flutter/lib/src/widgets/basic.dart]: re-run triggered...

**Comment by nayeemtby** [comment]
@flar I am new to both github and flutter and PRs, So is my work here done?

File: packages/flutter/lib/src/widgets/basic.dart
Comment (nayeemtby) [reply packages/flutter/lib/src/widgets/basic.dart]: Sorry, but I am new to all these, So is everything done?

Comment (flar) [reply packages/flutter/lib/src/widgets/basic.dart]: Sorry, everything is not yet done.

The tests are now all passing and you have your approving review so the PR is ready to submit. But, the tree is RED right now which means some other recently merged PR caused a post-submit test failure. It is inappropriate to push PRs unless the tree is GREEN.

Fortunately, there is a handy label you can add to this PR called "Waiting for tree to go green" which will auto-merge it when the tree clears up.

Let me know if you need help with finding and adding the label.

Comment (nayeemtby) [reply packages/flutter/lib/src/widgets/basic.dart]: I don't think I can add labels, so yes please 😊

**Pull request merged**
